src/psyclone/domain/lfric/lfric_arg_descriptor.py

```diff
@@ -316,9 +317,9 @@ def _validate_field(self, arg_type):
                 format(LFRicArgDescriptor.VALID_FIELD_NAMES,
                        self._nargs, arg_type))
 
-        # Field data_type is "real" for now, but will be determined by
+        # Field data_type is "gh_real" for now, but will be determined by
         # metadata descriptor as the second argument in issue #817
-        self._data_type = "real"
+        self._data_type = LFRicArgDescriptor.VALID_ARG_DATA_TYPES[0]
```

Comment:
I think you need to stick with "gh_real" here rather than just arbitrarily index into the list of valid arg types. This is OK as it's a stepping stone towards #817 as you already say in the comment.

Reply:
Done.